### PR TITLE
EWPP-3855: Support D10.2 and drop D9.

### PR DESCRIPTION
**Pull request by nagyad**

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -42,6 +42,19 @@ services:
       - NODE_MAX_SESSION=5
 
 pipeline:
+  composer-change-patch:
+    group: warmup
+    image: registry.fpfis.eu/fpfis/httpd-php:${PHP_VERSION}-ci
+    volumes:
+      - /cache:/cache
+    commands:
+      # Change the patch for 10.2.x.
```
Comment (brummbar) [inline .drone.yml]: Let's do the opposite, add 10.2.x patch in composer.json and replace it for 10.1. We install by default on 10.2 from the composer.json of the component.

**Review by brummbar — CHANGES_REQUESTED** [review]

**Review by drishu — DISMISSED** [review]

**Comment by miguelst** [comment]
Hello, 

We need to upgrade urgently our website to Drupal 10, but the upgrade is blocked by this component as it does not support Drupal 10.2 version, and with previous version the upgrade also is blocked due other dependencies. 
Could you provide a estimation about when the component will support it?

Thanks,
Miguel

File: composer.json [hunk 2]
```diff
@@ -80,7 +80,7 @@
         },
         "patches": {
             "drupal/core": {
-                "https://www.drupal.org/project/drupal/issues/2230909": "https://www.drupal.org/files/issues/2022-07-26/2230909-269.patch"
+                "https://www.drupal.org/project/drupal/issues/2230909": "https://www.drupal.org/files/issues/2023-12-21/2230909-309.patch"
```
Comment (brummbar) [inline composer.json]: We need to warn people that this patch is not working for Drupal 10.1.
```suggestion
                "Drupal 10.2.x only - see README.md for 10.1.x - https://www.drupal.org/project/drupal/issues/2230909": "https://www.drupal.org/files/issues/2023-12-21/2230909-309.patch"
```

File: README.md [hunk 3]
```diff
@@ -23,9 +23,18 @@ Check out the [README](/modules/oe_content_entity/README.md) of the module.
...
+* [PHP >=8.1](http://php.net/)
 * Virtuoso (or equivalent) triple store which contains the RDF representations of the following Publications Office (OP) vocabularies: Corporate Bodies, Target Audiences, Organisation Types, Resource Types, Eurovoc
 
+## Usage with Drupal 10.1
+
+For Drupal 10.1, the drupal/core patch needs to be overridden to the new one specifically made for 10.1.
+If you require this module in Drupal 10.1 you must set the patch in your composer.json.\
+You can do this with the following command:
+```bash
+composer config --merge --json "extra.patches.drupal/core" '{"https://www.drupal.org/project/drupal/issues/2230909": "https://www.drupal.org/files/issues/2022-07-26/2230909-269.patch"}'
```
Comment (brummbar) [inline README.md:35]: This is ok for our drone, but users of this module need an additional command:
```
composer config --merge --json "extra.patches-ignore.openeuropa/oe_content" '{"drupal/core": {"Drupal 10.2.x only - see README.md for 10.1.x - https://www.drupal.org/project/drupal/issues/2230909": "https://www.drupal.org/files/issues/2023-12-21/2230909-309.patch"}}'
```
They need to ignore the patch provided by oe_content and add the other.

**Review by brummbar — CHANGES_REQUESTED** [review]

File: modules/oe_content_project/src/Drush/Commands/ProjectCommands.php [hunk 4]
```diff
@@ -54,11 +54,27 @@ public function __construct(EntityTypeManagerInterface $entityTypeManager, Messe
...
+      $container->get('entity_type.manager'),
+      $container->get('messenger'),
+    );
+  }
+
+  /**
+   * Copies the project budget values from the old field to the new field.
    */
-  public function copyFieldValues(): void {
+  #[CLI\Command(name: 'oe_content_project:copy-budget-field-values', aliases: [])]
```
Comment (brummbar) [inline modules/oe_content_project/src/Drush/Commands/ProjectCommands.php]: We cannot change the name of the command.

**Review by brummbar — CHANGES_REQUESTED** [review]

**Review by brummbar — DISMISSED** [review]

**Review by brummbar — APPROVED** [review]

**Pull request merged**